src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/SafeEvpPKeyHandle.OpenSsl.cs

```diff
@@ -243,6 +243,13 @@ public static SafeEvpPKeyHandle OpenKeyFromProvider(string providerName, string
                 throw new PlatformNotSupportedException(SR.PlatformNotSupported_CryptographyOpenSSL);
             }
 
+            // 3.0+ are M_NN_00_PP_p (Major, Minor, 0, Patch, Preview)
+            // 1.x.y are 1_XX_YY_PP_p
+            if (SafeEvpPKeyHandle.OpenSslVersion < 0x3_00_00_00_0)
+            {
+                throw new PlatformNotSupportedException(SR.PlatformNotSupported_CryptographyOpenSSL3NotFound);
+            }
+
```

Comment:
Version checks always feel a bit hacky.  Can we do this in a different way?

Reply:
The way we usually do this would be to change `CryptoNative_LoadKeyFromProvider` to have a `int32_t** supported` parameter and set it appropriately in the code paths that already exist for Providers not being supported.

Reply:
We already do that elsewhere: https://github.com/dotnet/runtime/blob/fe0bad475bf8fa64818d973bb8e4124f09cadcee/src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/OpenSslX509ChainProcessor.cs#L1409 - what's the benefit of doing this in native code?

Reply:
I guess the osx build errors explain the "why" part